cpp/src/c_api/replicate_edgelist.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cugraph_c/algorithms.h>
+#include <cugraph_c/graph_generators.h>
+
+#include <c_api/abstract_functor.hpp>
+#include <c_api/capi_helper.hpp>
+#include <c_api/graph.hpp>
+#include <c_api/induced_subgraph_result.hpp>
+#include <c_api/resource_handle.hpp>
+#include <c_api/utils.hpp>
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/detail/shuffle_wrappers.hpp>
+#include <cugraph/detail/utility_wrappers.hpp>
+#include <cugraph/graph_functions.hpp>
+
+namespace {
+
+template <typename vertex_t, typename weight_t>
+cugraph_error_code_t dummy_function(
+  raft::handle_t const& handle,
+  cugraph::c_api::cugraph_type_erased_device_array_view_t const* src,
+  cugraph::c_api::cugraph_type_erased_device_array_view_t const* dst,
+  cugraph::c_api::cugraph_type_erased_device_array_view_t const* weights,
+  cugraph::c_api::cugraph_induced_subgraph_result_t** result,
+  cugraph::c_api::cugraph_error_t** error)
+{
+  return CUGRAPH_SUCCESS;
+}
+
+template <typename vertex_t, typename weight_t>
```

Comment:
With the change proposed below, this can become a functor like the other C API implementations.  The templating can occur on the `()` operator.

Reply:
Right. I created a functor instead that matches other C API implementations